File: CoderDojoJPA/src/main/java/com/skilldistillery/coderdojo/entities/Location.java
package com.skilldistillery.coderdojo.entities;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.OneToOne;

@Entity
public class Location {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private int id;
	@Column(name="name")
	private String name;
	@Column(name="details")
	private String details;
	
	@OneToOne
	@JoinColumn(name="address_id")
	private Address address;

}

File: coderDojoApp/src/app/models/location.ts
import { Address } from './address';

export class Location {
  id: number;
  name: string;
  details: string;
  address: Address;
 constructor(
  id?: number ,
  name: string = '',
  details: string = '',
  address?: {
    id?: number
  }){
    this.id = id;
    this.name = name;
    this.details = details;
    this.address = Object.assign(this, this.address);

  }
}
